tests/ert/unit_tests/resources/test_run_flow_simulator.py

```diff
@@ -112,15 +124,143 @@ def test_flowrunner_will_raise_on_unknown_version():
+def test_numcpu_maps_to_mpi_processes_with_flow(num_cpu):
+    Path("flow.ert").write_text(
+        dedent(f"""
+    NUM_REALIZATIONS 1
+    ECLBASE EIGHTCELLS
+    DATA_FILE EIGHTCELLS.DATA
+    RUNPATH realization-<IENS>
+    NUM_CPU {num_cpu}
+    FORWARD_MODEL FLOW
+    """).strip(),
+        encoding="utf-8",
     )
-    run_reservoirsimulator.RunReservoirSimulator(
-        "flow", FLOW_VERSION, "SPE1_PARALLEL.DATA"
-    ).run_flow()
+
+    run_cli(TEST_RUN_MODE, "--disable-monitoring", "flow.ert")
+    flow_stdout = Path("realization-0/FLOW.stdout.0").read_text(encoding="utf-8")
+    assert re.search(rf"Number of MPI processes:\s+{num_cpu}", flow_stdout)
+    assert re.search(r"Threads per MPI process:\s+1", flow_stdout)
+
+
+@pytest.mark.integration_test
+@pytest.mark.usefixtures("eightcells")
+@pytest.mark.skipif(not shutil.which("flowrun"), reason="flowrun not available")
+def test_user_can_specify_threads_and_oversubscribe_compute_node():
+    Path("flow.ert").write_text(
+        dedent("""
+    NUM_REALIZATIONS 1
+    ECLBASE EIGHTCELLS
+    DATA_FILE EIGHTCELLS.DATA
+    RUNPATH realization-<IENS>
+    NUM_CPU 1
+    FORWARD_MODEL FLOW(<OPTS>="--threads 2")  -- OPTS is parsed by flowrun
+    """).strip(),
+        encoding="utf-8",
+    )
+
+    run_cli(TEST_RUN_MODE, "--disable-monitoring", "flow.ert")
+    flow_stdout = Path("realization-0/FLOW.stdout.0").read_text(encoding="utf-8")
+    assert re.search(r"Number of MPI processes:\s+1", flow_stdout)
+    assert re.search(r"Threads per MPI process:\s+2", flow_stdout)
+
+
+@pytest.mark.integration_test
+@pytest.mark.usefixtures("eightcells")
+@pytest.mark.skipif(not shutil.which("flowrun"), reason="flowrun not available")
+def test_user_can_specify_threads_and_mpi_processes_and_oversubscribe_compute_node():
+    Path("flow.ert").write_text(
+        dedent("""
+    NUM_REALIZATIONS 1
+    ECLBASE EIGHTCELLS
+    DATA_FILE EIGHTCELLS.DATA
+    RUNPATH realization-<IENS>
+    NUM_CPU 1
+    FORWARD_MODEL FLOW(<OPTS>="--np 2 --threads 2")  -- OPTS is parsed by flowrun
+    """).strip(),
+        encoding="utf-8",
+    )
+
+    run_cli(TEST_RUN_MODE, "--disable-monitoring", "flow.ert")
+    flow_stdout = Path("realization-0/FLOW.stdout.0").read_text(encoding="utf-8")
+    assert re.search(r"Number of MPI processes:\s+2", flow_stdout)
+    assert re.search(r"Threads per MPI process:\s+2", flow_stdout)
+
+
+@pytest.mark.integration_test
+@pytest.mark.usefixtures("eightcells")
+@pytest.mark.skipif(not shutil.which("flowrun"), reason="flowrun not available")
+def test_setenv_can_be_used_to_set_threads():
+    Path("flow.ert").write_text(
+        dedent("""
+    NUM_REALIZATIONS 1
+    ECLBASE EIGHTCELLS
+    DATA_FILE EIGHTCELLS.DATA
+    RUNPATH realization-<IENS>
+    SETENV OMP_NUM_THREADS 2
+    NUM_CPU 1
+    FORWARD_MODEL FLOW()
+    """).strip(),
+        encoding="utf-8",
+    )
+
+    run_cli(TEST_RUN_MODE, "--disable-monitoring", "flow.ert")
```

Comment:
Since this runs the cli, I think it makes sense to move this to `tests/ert/ui_tests/cli/test_run_flow_simulator.py`

Reply:
Moved